src/com/invoca/docker/Image.groovy

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/groovy
+package com.invoca.docker
+
+class Image implements Serializable {
+  public static String LABEL_SCHEMA_VERSION = "org.label-schema.schema-version"
+  public static String LABEL_BUILD_DATE = "org.label-schema.build-date"
+  public static String LABEL_VCS_URL = "org.label-schema.vcs-url"
+
+  private String imageName
+  private String[] tags
+  private String baseDir
+  private Script script
+  private String[] imageNameWithTags
+  private String[] labels
+
+  public Image(Script script, String imageName, String tag, String baseDir = ".") {
+    this(script, imageName, [tag], baseDir)
+  }
+
+  public Image(Script script, String imageName, ArrayList tags, String baseDir = ".") {
+    this(script, imageName, tags as String[], baseDir)
+  }
+
+  public Image(Script script, String imageName, String[] tags, String baseDir = ".") {
+    this.script = script
+    this.imageName = imageName
+    this.tags = tags
+    this.baseDir = baseDir
+  }
+
+  public Image build(Map args) {
+    this.imageNameWithTags = buildImageNameWithTags()
+
+    def gitUrl = args.gitUrl
+    def buildArgs = args.buildArgs ?: [:]
+    def dockerFile = args.dockerFile ?: "Dockerfile"
```

Comment:
The way I read the README example was that args aren't named and are instead positional. Could you point out what I'm missing / misreading?

Reply:
The constructor arguments are positional, not the arguments to `Image#build`.  `build()` uses the Map -> named parameter magic in Groovy, similar to passing a hash as the last argument to a method in Ruby.

Since the majority of the arguments to `build` are optional, I felt it made more sense to use the named parameters in this case.  I'm not opposed to changing it for consistency though.

Reply:
I think the named args are ideal. Thanks for clarifying.